perfkitbenchmarker/linux_benchmarks/gpu_pcie_bandwidth_benchmark.py

```diff
@@ -0,0 +1,223 @@
+    and host to device):
+      * mean
+      * min
+      * max
+      * stddev
+  """
+  results = []
+  for metric in BENCHMARK_METRICS:
+    sequence = [x[metric] for x in result_dicts]
+    results.append(sample.Sample(
+        metric + ', min', min(sequence), 'MB/s', metadata))
+    results.append(sample.Sample(
+        metric + ', max', max(sequence), 'MB/s', metadata))
+    results.append(sample.Sample(
+        metric + ', mean', numpy.mean(sequence), 'MB/s', metadata))
+    results.append(sample.Sample(
+        metric + ', stddev', numpy.std(sequence), 'MB/s', metadata))
+  return results
+
+
+def _SetAndConfirmGpuClocks(benchmark_spec):
+  """Sets and confirms the GPU clocks with the values provided
+     in the gpu_pcie_bandwidth_clock_speeds flag. If a device
+     is queried and its clock speed does not allign with what
+     it was just set to, an expection will be raised.
+
+     Raises:
+      Exception if a GPU did not accept the provided clock speeds.
+  """
+  vm = benchmark_spec.vms[0]
+  desired_memory_clock = FLAGS.gpu_pcie_bandwidth_clock_speeds[0]
+  desired_graphics_clock = FLAGS.gpu_pcie_bandwidth_clock_speeds[1]
+  cuda_toolkit_8.SetGpuClockSpeed(vm,
+                                  desired_memory_clock,
+                                  desired_graphics_clock)
+  num_gpus = cuda_toolkit_8.QueryNumberOfGpus(vm)
+  for i in range(num_gpus):
+    if cuda_toolkit_8.QueryGpuClockSpeed(vm, i) != (desired_memory_clock,
+                                                    desired_graphics_clock):
+      raise Exception("Unrecoverable error setting "
+                      "GPU #{} clock speed to {},{}"
+                      .format(i, desired_memory_clock, desired_graphics_clock))
+
+
+
+def Run(benchmark_spec):
+  """Sets the GPU clock speed and runs the CUDA PCIe benchmark.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  vm = benchmark_spec.vms[0]
+  # Note:  The clock speed is set in this function rather than Prepare()
+  # so that the user can perform multiple runs with a specified
+  # clock speed without having to re-prepare the VM.
+  _SetAndConfirmGpuClocks(benchmark_spec)
+  iterations = FLAGS.gpu_pcie_bandwidth_iterations
+  samples = []
+  metadata = {}
+  metadata['iterations'] = iterations
+  metadata['clock_speeds'] = [FLAGS.gpu_pcie_bandwidth_clock_speeds[0],
+                              FLAGS.gpu_pcie_bandwidth_clock_speeds[1]]
+  run_command = '%s/extras/demo_suite/bandwidthTest --device=all' %\
+      cuda_toolkit_8.CUDA_TOOLKIT_INSTALL_DIR
+  for i in range(iterations):
+    stdout, _ = vm.RemoteCommand(run_command, should_log=True)
+    samples.append(_ParseOutputFromSingleIteration(stdout))
+    if 'device_info' not in metadata:
+      metadata['device_info'] = _ParseDeviceInfo(stdout)
+  return _CalculateMetricsOverAllIterations(samples, metadata)
```

Comment:
I think we should also publish raw data for each iteration. So we can calculate different percentiles across different runs (on different vms).